src/cli/java/org/commcare/util/screen/MultiSelectEntityScreen.java

```diff
@@ -0,0 +1,84 @@
+package org.commcare.util.screen;
+
+import static org.commcare.session.SessionFrame.STATE_MULTIPLE_DATUM_VAL;
+
+import org.commcare.core.interfaces.EntitiesSelectionCache;
+import org.commcare.modern.session.SessionWrapper;
+import org.commcare.session.CommCareSession;
+import org.commcare.suite.model.MultiSelectEntityDatum;
+import org.javarosa.core.model.instance.TreeReference;
+import org.javarosa.core.util.PropertyUtils;
+
+import java.sql.SQLException;
+
+import javax.annotation.Nullable;
+
+/**
+ * Variation of EntityScreen to allow for selection of multiple entities at once
+ */
+public class MultiSelectEntityScreen extends EntityScreen {
+
+    private int maxSelectValue = -1;
+
+    private final EntitiesSelectionCache entitiesSelectionCache;
+
+    public MultiSelectEntityScreen(boolean handleCaseIndex, boolean full,
+            SessionWrapper session, @Nullable String[] selectedValues,
+            EntitiesSelectionCache entitiesSelectionCache)
+            throws CommCareSessionException, SQLException {
+        super(handleCaseIndex, full, session);
+        this.entitiesSelectionCache = entitiesSelectionCache;
+        processSelectedValues(selectedValues, session);
+    }
+
+    private void processSelectedValues(String[] selectedValues, SessionWrapper session)
+            throws SQLException, CommCareSessionException {
+        if (selectedValues != null) {
+
+            String[] evaluatedValues = new String[selectedValues.length];
+            for (int i = 0; i < selectedValues.length; i++) {
+                TreeReference currentReference = getEntityReference(selectedValues[i]);
+                if (currentReference == null) {
```

Comment:
Would it be worth gathering up all erroring entities instead of stopping on the first one?

Reply:
I could not think of a compelling reason to do so, I think just the single entity should be sufficient in order to point out what's wrong to a developer. For users, Showing all those entities can be overwhelming and cluttery when there are more of those present.

Reply:
sounds good